signer/src/transaction_coordinator.rs

```diff
@@ -296,9 +297,15 @@ where
             return Ok(());
         };
 
+        // Construct the transaction package and store it in the database.
         let transaction_package = pending_requests.construct_transactions()?;
 
         for mut transaction in transaction_package {
+            // Store the transaction in the database before we broadcast.
+            transaction
+                .store_as_sweep_transaction(&self.context.get_storage_mut(), bitcoin_chain_tip)
+                .await?;
```

Comment:
Doesn't this mean that the coordinator stores it at a different time compared to the other signers?

Reply:
Database time i.e. `created_at`? Yeah, but the time is largely useless.

Reply:
Well I guess I'm thinking that something can fail between now and when the coordinator informs the other signers that they are going to (or have) broadcast the transaction.

Reply:
For #709 I think I'm planning on the broadcast to other signers happening directly after this line, and if the broadcast itself actually fails for some reason then the signers will have an extra "sweep transaction" in their db, but it'll never match any transactions in blocks.

Or do you think this should be closer to the end of `sign_and_broadcast()` right before the actual broadcast?

Reply:
Oh won't the tx signer receive this message then? I'm think the signer always reads the tx coordinators messages and does the writing. I'm thinking that it'll be simpler that way. Thoughts?

Reply:
I honestly hadn't thought through that yet, but yeah if we go that route then this line will just be moved to the signer and become the msg send instead.